### PR TITLE
Report when backoff limit is exceeded

### PR DESCRIPTION
**Pull request by fao89**

### CODE REVIEW

**Review by jpodivin — COMMENTED** [review]
Is there going to be a follow up removing now no longer useful constants?

**Review by jpodivin — COMMENTED** [review]
Is the `Failed` going to be used in any other way besides providing information if one calls up the resource from client? Any plans for it to be involved in operator behavior for example?

**Comment by fao89** [comment]
> Is the `Failed` going to be used in any other way besides providing information if one calls up the resource from client? Any plans for it to be involved in operator behavior for example?

yes, it will be used here: https://github.com/openstack-k8s-operators/dataplane-operator/pull/879/files for tracking the attempts on the dataplane-operator side

File: controllers/openstack_ansibleee_controller.go [hunk 1]
```diff
@@ -208,22 +208,30 @@ func (r *OpenStackAnsibleEEReconciler) Reconcile(ctx context.Context, req ctrl.R
...
-			condition.ErrorReason,
-			condition.SeverityWarning,
-			ansibleeev1.AnsibleExecutionJobErrorMessage,
+			condition.JobReadyCondition,
+			errorReason,
+			severity,
+			condition.JobReadyErrorMessage,
 			err.Error()))
 		instance.Status.JobStatus = ansibleeev1.JobStatusFailed
+		instance.Status.Failed = ansibleeeJob.GetTotalFailedAttempts()
```
Comment (rabi) [inline controllers/openstack_ansibleee_controller.go:234]: What scenarios pods won't be retried, so we get the value here different from backoffLimit we set the OpenStackDataPlaneDeployment(job) spec (default 6). Just wondering if there is value in keeping track of this here unless we allow to changing `podFailurePolicy` for jobs?

Comment (fao89) [reply controllers/openstack_ansibleee_controller.go:234]: this is for tracking each attempt:
attempt 1 failed
attempt 2 failed
...
the idea is to get more transparency on the retries: [OSPRH-6593](https://issues.redhat.com/browse/OSPRH-6593)
https://github.com/openstack-k8s-operators/dataplane-operator/pull/879/files#r1611441361

Comment (rabi) [reply controllers/openstack_ansibleee_controller.go:234]: Why would we want to add every retry failure to the status unless it has some details that's different from every retry?

Comment (fao89) [reply controllers/openstack_ansibleee_controller.go:234]: I think exposing the retry number shows some progress

Comment (jpodivin) [reply controllers/openstack_ansibleee_controller.go:234]: I understand the benefit, but I'm unsure if status is the right place to put it.

In principle it makes sense to let people know how many attempts have been made. No doubt about that. But my understanding of `status` is that it isn't concerned that much about history, as it is about ... well *state*. 

Personally I wouldn't agonize over it. At most we are talking about style and practice, not about functionality.

Comment (fao89) [reply controllers/openstack_ansibleee_controller.go]: I've dropped the `failed` field

**Comment by fao89** [comment]
recheck

**Comment by fao89** [comment]
recheck

**Review by slagle — APPROVED** [review]

**Pull request merged**